examples/BasicSerialLogger/BasicSerialLogger.ino

```diff
@@ -0,0 +1,250 @@
+   rf24Logger.debug((const __FlashStringHelper*) vendorID,
+         "RAM message with %S", (const __FlashStringHelper*) globalProgmemText);
+   rf24Logger.trace((const __FlashStringHelper*) vendorID,
+         "RAM message with %S", (const __FlashStringHelper*) globalProgmemText);
+
+   Serial.println();
+}
+
+void logRamMessageWithFMacroStringArgument()
+{
+   rf24Logger.error((const __FlashStringHelper*) vendorID,
+         "RAM message with %S", F("F macro string 1"));
+   rf24Logger.warn((const __FlashStringHelper*) vendorID, "RAM message with %S",
+         F("F macro string 2"));
+   rf24Logger.info((const __FlashStringHelper*) vendorID, "RAM message with %S",
+         F("F macro string 3"));
+   rf24Logger.debug((const __FlashStringHelper*) vendorID,
+         "RAM message with %S", F("F macro string 4"));
+   rf24Logger.trace((const __FlashStringHelper*) vendorID,
+         "RAM message with %S", F("F macro string 5"));
+
+   Serial.println();
+}
+
+void logProgmemMessageWithRamStringArgument()
+{
+   rf24Logger.error((const __FlashStringHelper*) vendorID,
+         (const __FlashStringHelper*) globalProgmemMessageWithRamString,
+         "RAM string 1");
+   rf24Logger.warn((const __FlashStringHelper*) vendorID,
+         (const __FlashStringHelper*) globalProgmemMessageWithRamString,
+         "RAM string 2");
+   rf24Logger.info((const __FlashStringHelper*) vendorID,
+         (const __FlashStringHelper*) globalProgmemMessageWithRamString,
+         "RAM string 3");
+   rf24Logger.debug((const __FlashStringHelper*) vendorID,
+         (const __FlashStringHelper*) globalProgmemMessageWithRamString,
+         "RAM string 4");
+   rf24Logger.trace((const __FlashStringHelper*) vendorID,
+         (const __FlashStringHelper*) globalProgmemMessageWithRamString,
+         "RAM string 5");
+
+   Serial.println();
+}
+
+void logMessageWithAdditionalTagAtTheBeginning()
+{
+   rf24Logger.info((const __FlashStringHelper*) vendorID,
+         "%S info log with additional tag at the beginning", F("RF24Log.cpp"));
+
+   Serial.println();
+}
+
+void logMessageWithUnknownFormat()
+{
+   rf24Logger.info((const __FlashStringHelper*) vendorID,
+         "info log with unknown format   : %p", F("flash text"));
+
+   Serial.println();
+}
+
+void logFloatNumber()
+{
+   rf24Logger.info((const __FlashStringHelper*) vendorID,
+         " info log with double value %D", 3.14);
+
+   Serial.println();
+
+   rf24Logger.warn((const __FlashStringHelper*) vendorID,
+         " warn log with double value %D", 3.14);
+
+   Serial.println();
+
+   rf24Logger.error((const __FlashStringHelper*) vendorID,
+         "error log with double value %D", 3.14);
+
+   Serial.println();
+
+   rf24Logger.debug((const __FlashStringHelper*) vendorID,
+         "debug log with double value %F", 2.71);
+
+   Serial.println();
+
+   rf24Logger.trace((const __FlashStringHelper*) vendorID,
+         "trace log with double value %F", 2.71);
+
+   Serial.println();
+}

```

Comment:
I know this is more of a test, but this kind of repetitive example would be off-putting to beginners (when we're ready to call this lib stable)

Reply:
Maybe the debug message should contain the newline, it would look a tad cleaner?

Reply:
I think the messages shouldn't automatically append the newline. Logs should go direct line by line. I don't even remember why I put those ```Serial.println()```.

In my opinion - what @2bndy5 meant - is that the construction of the method (repetitive output) may be hard to understand and a bit boring. The user need to deep analyze line by line to see the differencies. Maybe the examples should be just reworked to show in an easy way the potential of the library.